pallets/settlement/src/lib.rs

```diff
@@ -699,7 +701,27 @@ decl_module! {
             // Authorize the instruction.
             Self::unsafe_authorize_instruction(did, instruction_id, portfolios_set)?;
 
-            MercatTxDataStorage::append(instruction_id, data);
+            MercatTxDataStorage::append(instruction_id, data.clone());
+
+            // Update the transaction sender's ordering state.
+            if let MercatTxData::InitializedTransfer(tx_data) =  data {
+                let result = base64::decode(&tx_data);
```

Comment:
It's possible to do without cloning `data` if the following reordering is acceptable:
```
            // Update the transaction sender's ordering state.
            if let MercatTxData::InitializedTransfer(tx_data) = &data {
                let result = base64::decode(tx_data);

                // the rest of the code block

            }
            MercatTxDataStorage::append(instruction_id, data);
```
That is, cloning `data` is not required to obtain `result`.